android/src/main/java/com/bitmovin/player/reactnative/RNPlayerViewManager.kt

```diff
@@ -264,11 +268,25 @@ class RNPlayerViewManager(private val context: ReactApplicationContext) : Simple
                     Log.e(MODULE_NAME, "Cannot create a PlayerView, because no activity is attached.")
                     return@post
                 }
-                val playerView = PlayerView(
-                    currentActivity,
-                    player,
-                    view.config?.playerViewConfig ?: PlayerViewConfig(),
-                )
+                val rnPlayerConfigWrapper = view.config
+                    ?: playerConfig
+                        ?.let { JsonConverter.toRNPlayerViewConfigWrapper(it) }
+                val nativePlayerViewConfig = rnPlayerConfigWrapper?.playerViewConfig ?: PlayerViewConfig()
+
+                val playerView: PlayerView
+                if (nativePlayerViewConfig.uiConfig == UiConfig.Disabled ||
+                    (rnPlayerConfigWrapper?.userInterfaceType ?: UserInterfaceType.Bitmovin) ==
+                    UserInterfaceType.Bitmovin
+                ) {
+                    playerView = PlayerView(currentActivity, player, nativePlayerViewConfig)
+                } else {
+                    val overwrittenConfig = nativePlayerViewConfig.copy(uiConfig = UiConfig.Disabled)
+                    playerView = PlayerView(currentActivity, player, overwrittenConfig)
+                    val subtitleView = SubtitleView(currentActivity)
+                    subtitleView.setPlayer(player)
+                    view.addSubtitleView(subtitleView)
+                }
```

Comment:
For me this is very hard to read.
I am aware of the limitation we stated on the `PlayerViewConfig.uiConfig`
> Configuring the `uiConfig` only has an effect if the {@link StyleConfig.userInterfaceType} is set to {@link UserInterfaceType.Bitmovin}.

And that we have to take care of this now on android, as the `UserInterfaceType` is now also valid for android.

Maybe the (in another comment suggested) split of the `RNPlayerViewConfig` and the `userInterfaceType` makes this already clearer.

Reply:
please take another look - there were multiple commits updating the readability of this logic